homeassistant/__main__.py

```diff
@@ -255,10 +255,13 @@ def closefds_osx(min_fd: int, max_fd: int) -> None:
 
 def cmdline() -> List[str]:
     """Collect path and arguments to re-execute the current hass instance."""
-    if sys.argv[0].endswith('/__main__.py'):
+    if sys.argv[0].endswith(os.path.sep + '__main__.py'):
```

Comment:
`os.path.basename(sys.argv[0]) == '__main__.py'` is a more natural way of checking the filename, I think.